### DISCUSSION TITLE
Is there a way to update a group plus dependencies without affecting direct requirements?

### DISCUSSION THREAD

**Discussion by rickselby** [ruby/rubygems#5603] [Q&A]
Given the Gemfile:

```ruby
group :test do
  gem 'activesupport'
end

gem 'i18n'
```

and the Gemfile.lock:

```
GEM
  remote: https://rubygems.org/
  specs:
    activesupport (7.0.0)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (>= 1.6, < 2)
      minitest (>= 5.1)
      tzinfo (~> 2.0)
    concurrent-ruby (1.1.10)
    i18n (1.9.1)
      concurrent-ruby (~> 1.0)
    minitest (5.14.4)
    tzinfo (2.0.4)
      concurrent-ruby (~> 1.0)

PLATFORMS
  x86_64-linux

DEPENDENCIES
  activesupport
  i18n

BUNDLED WITH
   2.3.15
```

Running `bundle update --group test` updates `activesupport`, `i18n` and `minitest`.
Running `bundle update --group test --conservative` updates `activesupport` only.

Is there a way of updating just `activesupport` and `minitest` but not touching `i18n` (because it's declared directly in the Gemfile)? i.e. _Update all gems in the group plus their dependencies, unless the dependencies are shared with gems defined outside of the group_

I think this is what `--conservative` used to do, but was changed in https://github.com/rubygems/rubygems/issues/4319.

**Comment by deivid-rodriguez** [comment]
Hi!

Thanks for your question. I _think_ `--conservative` was indeed documented as you mentioned, but I think it never actually did that? Did you try it or is it just a guess?

I guess to implement this feature we would need a separate flag, like `--lock-shared-dependencies`, or introduce `--conservative` levels, like `--conservative=all`, and `--conservative=shared-only` or something.

**Comment by deivid-rodriguez** [comment]
This is being tracked as an issue at https://github.com/rubygems/rubygems/issues/7117, so let's consolidate the discussion in there. You're right this used to work as you point out.
